### ISSUE TITLE
Object detector doesn't find matches

### ISSUE THREAD

**Issue by tagy** [apple/turicreate#3148]
Hi, I have setup the current version of turicreate in linux and tried training a model for object detection. The training completes but the model doesn't find objects in test images, either trying predict() or exporting the model to the sample iOS app.

I have checked the training images and annotations are correct using util.draw_bounding_boxes() and it looks good.

I also tried the same data in the MakeML app, that seems to use turicreate 5.2.1 the model from that works correctly with their test tool and also in iOS.

Does anyone have any clue how to debug this? When training the loss is decreasing by a similar amount as seen in the MakeML app. I would rather be able to run it on my own hardware. Thanks.

**Comment by tagy** [comment]
I have re-installed linux and stared again and now I can create the model and it does find matches, so must have been something I was doing wrong. You can close this. 

The model I am getting does not work as well as the one from MakeML though, I guess I could try installing turicreate 5.2.1 but that is a different question.

**Comment by TobyRoseman** [comment]
We have a serious issue with our TensorFlow implementation of object detection, this is what gets used on Linux. It's completely broken with our latest (6.2) release. If you have more than two classes in your dataset, then things have been broken since 6.0 (see #3031).

So if you have two or less classes/objects in your dataset, please use TuriCreate 6.1. If you have more than two classes, please use 5.8.

I am actively looking into these object detection issues. I'm planning to do a new release of TuriCreate once these issues are resolved.

**Comment by tagy** [comment]
I have installed 6.1 and it seems to be working as expected now (I have one object). Thanks.

**Comment by TobyRoseman** [comment]
We just released a point release (TuriCreate 6.2.2) with this bug fix. Upgrade your version of TuriCreate and you should be good to go.
